lib/constructs/interface.js

```diff
@@ -1136,49 +1141,55 @@ class Interface {
           throw new Error('Internal error: constructor ${this.name} is not installed on the passed global object');
         }
 
-        let obj = Object.create(ctor.prototype);
-        obj = exports.setup(obj, globalObject, constructorArgs, privateData);
-        return obj;
-      };
-      exports.createImpl = function createImpl(globalObject, constructorArgs, privateData) {
-        const obj = exports.create(globalObject, constructorArgs, privateData);
-        return utils.implForWrapper(obj);
-      };
-      exports._internalSetup = function _internalSetup(obj) {
+        let wrapperObject = Object.create(ctor.prototype);
+        exports._internalSetup(wrapperObject);
     `;
 
-    if (this.idl.inheritance) {
+    if (this.isLegacyPlatformObj) {
       this.str += `
-        ${this.idl.inheritance}._internalSetup(obj);
+        wrapperObject = new Proxy(wrapperObject, proxyHandler);
       `;
     }
 
-    this.generateOnInstance();
-
     this.str += `
+        implObject[wrapperSymbol] = wrapperObject;
+        wrapperObject[impl] = implObject;
+        return wrapperObject;
       };
-      exports.setup = function setup(obj, globalObject, constructorArgs = [], privateData = {}) {
-        privateData.wrapper = obj;
+      exports.create = function create(globalObject, constructorArgs, privateData) {
+        const implObject = exports.createImpl(globalObject, constructorArgs, privateData);
+        return utils.wrapperForImpl(implObject);
+      };
+      exports.createImpl = function createImpl(globalObject, constructorArgs = [], privateData = {}) {
+        const implObject = new Impl.implementation(globalObject, constructorArgs, privateData);
 
-        exports._internalSetup(obj);
-        Object.defineProperty(obj, impl, {
-          value: new Impl.implementation(globalObject, constructorArgs, privateData),
-          configurable: true
-        });
+        implObject[wrapperSymbol] = null;
```

Comment:
Why&nbsp;not use&nbsp;a&nbsp;getter that&nbsp;turns&nbsp;into a&nbsp;data&nbsp;property?

That&nbsp;will&nbsp;remove the&nbsp;need for&nbsp;`createWrapperSymbol` and&nbsp;`globalObjectSymbol`.